src/patternfly/components/Nav/nav.scss

```diff
@@ -354,389 +253,290 @@
+  --pf-c-nav__link--hover--BackgroundColor: var(--pf-c-nav__tertiary-list__link--hover--BackgroundColor);
+  --pf-c-nav__link--focus--BackgroundColor: var(--pf-c-nav__tertiary-list__link--focus--BackgroundColor);
+  --pf-c-nav__link--active--BackgroundColor: var(--pf-c-nav__tertiary-list__link--active--BackgroundColor);
+  --pf-c-nav__link--m-current--BackgroundColor: var(--pf-c-nav__tertiary-list__link--m-current--BackgroundColor);
+  --pf-c-nav__link--before--BorderColor: var(--pf-c-nav__tertiary-list__link--before--BorderColor);
 
-      &::after {
-        background-color: var(--pf-c-nav__tertiary-list-link--focus--after--BackgroundColor);
-      }
-    }
-
-    &.pf-m-active,
-    &:active {
-      --pf-c-nav__tertiary-list-link--Color: var(--pf-c-nav__tertiary-list-link--active--Color);
-
-      &::after {
-        background-color: var(--pf-c-nav__tertiary-list-link--active--after--BackgroundColor);
-      }
-    }
+  .pf-c-nav__link:hover {
+    --pf-c-nav__link--before--BorderBottomWidth: var(--pf-c-nav__tertiary-list__link--hover--before--BorderWidth);
+  }
 
-    &.pf-m-current {
-      --pf-c-nav__tertiary-list-link--Color: var(--pf-c-nav__tertiary-list-link--m-current--Color);
+  .pf-c-nav__link:focus {
+    --pf-c-nav__link--before--BorderBottomWidth: var(--pf-c-nav__tertiary-list__link--focus--before--BorderWidth);
+  }
 
-      &::after {
-        background-color: var(--pf-c-nav__tertiary-list-link--m-current--after--BackgroundColor);
-      }
-    }
+  .pf-c-nav__link:active {
+    --pf-c-nav__link--before--BorderBottomWidth: var(--pf-c-nav__tertiary-list__link--active--before--BorderWidth);
+  }
 
-    // After - border bottom treatment
-    &::after {
-      position: absolute;
-      bottom: 0;
-      left: 0;
-      display: block;
-      width: 100%;
-      height: var(--pf-c-nav__tertiary-list-link--after--Height);
-      content: "";
-    }
+  .pf-c-nav__link.pf-m-current {
+    --pf-c-nav__link--before--BorderBottomWidth: var(--pf-c-nav__tertiary-list__link--m-current--before--BorderWidth);
   }
 }
 
 // Subnav
 .pf-c-nav__subnav {
-  flex: 1 0 100%;
+  --pf-c-nav__link--FontSize: var(--pf-c-nav__subnav__link--FontSize);
+  --pf-c-nav__simple-list__link--MarginTop: var(--pf-c-nav--subnav__simple-list__link--MarginTop);
+  --pf-c-nav__simple-list__link--MarginBottom: var(--pf-c-nav--subnav__simple-list__link--MarginBottom);
+  --pf-c-nav__simple-list__link--FontSize: var(--pf-c-nav__subnav__link--FontSize);
 
-  // Set default states
   max-height: 0;
-  margin-top: 0;
-  overflow: hidden;
-  opacity: 0;
+  padding-bottom: var(--pf-c-nav__subnav--PaddingBottom);
+  padding-left: var(--pf-c-nav__subnav--PaddingLeft);
   transition: var(--pf-c-nav--Transition);
 
-  // Hide scrollbars
-  &::-webkit-scrollbar {
-    display: none;
-    visibility: hidden;
-  }
-
-  // Expanded list item
   .pf-c-nav__item.pf-m-expanded & {
     max-height: var(--pf-c-nav__subnav--MaxHeight);
```

Comment:
This var name is probably better as `--pf-c-nav__item--m-expanded__subnav--MaxHeight` since `__subnav` actually has a `max-height`

And since you declare `max-height` for `__subnav`, you could use `--pf-c-nav__subnav--MaxHeight` there, then here declare `--pf-c-nav__subnav--MaxHeight: var(--pf-c-nav__item--m-expanded__subnav--MaxHeight);`